src/algebra/jordan/basic.lean

```diff
@@ -0,0 +1,213 @@
+  [cabreragarciarodriguezpalacios2014]
+* [Hanche-Olsen and Størmer, Jordan Operator Algebras][hancheolsenstormer1984]
+* [McCrimmon, A taste of Jordan algebras][mccrimmon2004]
+
+-/
+
+variables (A : Type*)
+
+/-- A (non-commutative) Jordan multiplication. -/
+class is_jordan [has_mul A] :=
+(lmul_comm_rmul : ∀ a b : A, (a * b) * a = a * (b * a))
+(lmul_lmul_comm_lmul : ∀ a b : A, (a * a) * (a * b) = a * ((a * a) * b))
+(lmul_lmul_comm_rmul : ∀ a b : A, (a * a) * (b * a) = ((a * a) * b) * a)
+(lmul_comm_rmul_rmul : ∀ a b : A, (a * b) * (a * a) = a * (b * (a * a)))
+(rmul_comm_rmul_rmul : ∀ a b : A, (b * a) * (a * a) = (b * (a * a)) * a)
+
+/-- A commutative Jordan multipication -/
+class is_comm_jordan [has_mul A] :=
+(mul_comm : ∀ a b : A, a * b = b * a)
+(jordan : ∀ a b : A, (a * b) * (a * a) = a * (b * (a *a)))
+
+/-- A (commutative) Jordan multiplication is also a Jordan multipication -/
+@[priority 100] -- see Note [lower instance priority]
+instance is_comm_jordan.to_is_jordan [has_mul A] [is_comm_jordan A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw [is_comm_jordan.mul_comm, is_comm_jordan.mul_comm a b],
+  lmul_lmul_comm_lmul := λ a b, by rw [is_comm_jordan.mul_comm (a * a) (a * b),
+    is_comm_jordan.jordan, is_comm_jordan.mul_comm b (a * a)],
+  lmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm, ←is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm],
+  lmul_lmul_comm_rmul :=  λ a b, by rw [is_comm_jordan.mul_comm (a * a) (b * a),
+    is_comm_jordan.mul_comm b a, is_comm_jordan.jordan, is_comm_jordan.mul_comm,
+    is_comm_jordan.mul_comm b (a * a)],
+  rmul_comm_rmul_rmul := λ a b, by rw [is_comm_jordan.mul_comm b a, is_comm_jordan.jordan,
+    is_comm_jordan.mul_comm], }
+
+
+/-- Semigroup multiplication satisfies the (non-commutative) Jordan axioms-/
+@[priority 100] -- see Note [lower instance priority]
+instance semigroup.is_jordan [semigroup A] : is_jordan A :=
+{ lmul_comm_rmul := λ a b, by rw mul_assoc,
+  lmul_lmul_comm_lmul := λ a b, by rw [mul_assoc, mul_assoc],
+  lmul_comm_rmul_rmul := λ a b, by rw [mul_assoc],
+  lmul_lmul_comm_rmul := λ a b, by rw [←mul_assoc],
+  rmul_comm_rmul_rmul := λ a b, by rw [← mul_assoc, ← mul_assoc], }
+
+@[priority 100] -- see Note [lower instance priority]
+instance comm_semigroup.is_comm_jordan [comm_semigroup A] : is_comm_jordan A :=
+{ mul_comm := mul_comm,
+  jordan := λ a b, mul_assoc _ _ _, }
+
+local notation `L` := add_monoid.End.mul_left
+local notation `R` := add_monoid.End.mul_right
+
+/-! The Jordan axioms can be expressed in terms of commuting multiplication operators -/
+section lie
+variables {A} [non_unital_non_assoc_ring A] [is_jordan A]
+
+@[simp] lemma lie_lmul_rmul (a : A) : ⁅L a, R a⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_comm_rmul _ _).symm
+
+@[simp] lemma lie_lmul_lmul_sq (a : A) : ⁅L a, L (a * a)⁆ = 0 :=
+add_monoid_hom.ext $ λ b, sub_eq_zero_of_eq (is_jordan.lmul_lmul_comm_lmul _ _).symm
```

Comment:
@ocfnash: given this section is about things commuting, what do you think of
```suggestion
@[simp] lemma commute_lmul_rmul (a : A) : commute (L a) (R a) = 0 :=
sorry

@[simp] lemma commute_lmul_lmul_sq (a : A) : commute (L a) (L (a * a)) :=
sorry
```

I assume we have something like `commute.bracket_eq_zero` somewhere?

Reply:
I'm on the fence. Given that we seem not to have `commute.bracket_eq_zero` I vote to leave these as they are and possibly return in a future PR.

Reply:
PR'd that lemma in #15675, we could easily copy those lemmas into this PR, they're not very deep in the import tree.

Reply:
@eric-wieser I take it you mean:
```
@[simp] lemma commute_lmul_rmul (a : A) : commute (L a) (R a) :=
add_monoid_hom.ext $ λ b, (is_jordan.lmul_comm_rmul _ _).symm
```
?

Reply:
Done.